components/graph/git-graph-actions.js

```diff
@@ -100,17 +99,17 @@ GraphActions.Reset.prototype.createHoverGraphic = function() {
   var nodes = context.node().getPathToCommonAncestor(remoteRef.node()).slice(0, -1);
   return new ResetViewModel(nodes);
 }
-GraphActions.Reset.prototype.perform = function(callback) {
+GraphActions.Reset.prototype.perform = function() {
   var self = this;
   var context = this.graph.currentActionContext();
   var remoteRef = context.getRemoteRef(self.graph.currentRemote());
-  components.create('yesnodialog', { title: 'Are you sure?', details: 'Resetting to ref: ' + remoteRef.name + ' cannot be undone with ungit.'})
+  return components.create('yesnodialog', { title: 'Are you sure?', details: 'Resetting to ref: ' + remoteRef.name + ' cannot be undone with ungit.'})
     .show()
     .closeThen(function(diag) {
       if (!diag.result()) return;
-      self.server.postPromise('/reset', { path: self.graph.repoPath(), to: remoteRef.name, mode: 'hard' })
-        .then(function() { context.node(remoteRef.node()); })
-    }).finally(callback)
+      return self.server.postPromise('/reset', { path: self.graph.repoPath(), to: remoteRef.name, mode: 'hard' })
+        .then(function() { context.node(remoteRef.node()); });
+    }).closePromise;
```

Comment:
That is why I suggested to return the close promise in `show` https://github.com/FredrikNoren/ungit/pull/852#discussion_r97205602 😉